terraform/main.tf

```diff
@@ -0,0 +1,284 @@
+# These are the supported environments. These need to map to the Terraform
+# workspaces.
+locals {
+  short_environment_names = {
+    staging    = "stg"
+  }
+  environment_names = {
+    staging    = "staging"
+  }
+}
+
+locals {
+  environment = lookup(local.environment_names, terraform.workspace)
+}
+
+locals {
+  app_name             = "record-childrens-vaccinations"
+  abbreviated_app_name = "rcv"
+  region               = "eu-west-2"
+  env                  = local.short_environment_names[local.environment]
+  image                = "public.ecr.aws/z8i3v8n4/record-childrens-vaccinations:latest"
+  db_port              = 5432
+  webapp_port          = 4000
+
+  azs                  = ["eu-west-2a", "eu-west-2b"]
+  private_subnets      = ["10.0.1.0/24", "10.0.2.0/24"]
+  public_subnets       = ["10.0.101.0/24", "10.0.102.0/24"]
+  database_subnets     = ["10.0.201.0/24", "10.0.202.0/24"]
+
+  tags                 = {
+    AppName     = local.app_name
+    Environment = local.environment
+    Source      = "terraform"
+  }
+}
+
+locals {
+  app_name_underscored = replace(local.app_name, "-", "_")
+}
+
+provider "aws" {
+  region = local.region
+}
+
+
+data "aws_secretsmanager_secret" "rails_master_key" {
+  name = "rails_master_key"
+}
+
+data "aws_secretsmanager_secret_version" "rails_master_key_latest" {
+  secret_id = data.aws_secretsmanager_secret.rails_master_key.id
+}
+
+data "aws_secretsmanager_secret" "support_user_credentials" {
+  name = "support_user_credentials"
+}
+
+data "aws_secretsmanager_secret_version" "support_user_credentials_latest" {
+  secret_id = data.aws_secretsmanager_secret.support_user_credentials.id
+}
+
+data "aws_secretsmanager_secret_version" "db_credentials_latest" {
+  secret_id = "db_credentials"
+}
+
+locals {
+  db_credentials = jsondecode(data.aws_secretsmanager_secret_version.db_credentials_latest.secret_string)
+  db_username    = local.db_credentials.username
+  db_password    = local.db_credentials.password
+}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+
+  name = "${local.abbreviated_app_name}-${local.env}"
+  cidr = "10.0.0.0/16"
+
+  azs               = local.azs
+  private_subnets   = local.private_subnets
+  public_subnets    = local.public_subnets
+  database_subnets  = local.database_subnets
+  create_database_subnet_group = true
+
+  enable_nat_gateway = true
+  single_nat_gateway  = true
+
+  tags = local.tags
+}
+
+module "db" {
+  source  = "terraform-aws-modules/rds/aws"
+
+  identifier = "${local.abbreviated_app_name}-${local.env}-db"
+  apply_immediately = true
+
+  engine            = "postgres"
+  engine_version    = "13.7"
+  family            = "postgres13"
+  instance_class    = "db.t3.micro"
```

Comment:
Will be interesting to see if we need to revisit this for prod, but I assume even a micro is plenty? Struggling to find exact specs.